org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/DiagnosticsHandler.java

```diff
@@ -119,12 +126,49 @@ public void beginReporting() {
 
 	@Override
 	public void endReporting() {
-		JavaLanguageServerPlugin.logInfo(problems.size() + " problems reported for " + this.uri.substring(this.uri.lastIndexOf('/')));
+		List<IProblem> allProblems = problems;
+		if (nonProjectFile) {
+			allProblems = new ArrayList<>();
+			allProblems.add(createNonProjectProblem());
+			allProblems.addAll(problems);
+		}
+		JavaLanguageServerPlugin.logInfo(allProblems.size() + " problems reported for " + this.uri.substring(this.uri.lastIndexOf('/')));
 		boolean isDiagnosticTagSupported = JavaLanguageServerPlugin.getPreferencesManager().getClientPreferences().isDiagnosticTagSupported();
-		PublishDiagnosticsParams $ = new PublishDiagnosticsParams(ResourceUtils.toClientUri(uri), toDiagnosticsArray(this.cu, problems, isDiagnosticTagSupported));
+		PublishDiagnosticsParams $ = new PublishDiagnosticsParams(ResourceUtils.toClientUri(uri), toDiagnosticsArray(this.cu, allProblems, isDiagnosticTagSupported));
 		this.connection.publishDiagnostics($);
 	}
 
+	private IProblem createNonProjectProblem() {
+		String fileName = cu.getElementName();
+		String projectName = cu.getJavaProject().getProject().getName();
+		String message = null;
+		int problemId = NON_PROJECT_JAVA_FILE;
+		if (shouldReportAllErrors()) {
+			if (isDefaultProject) {
+				message = "File " + fileName + " is non-project file, only JDK is added to the buildpath";

```

Comment:
fileName + " is a non-project file, only JDK classes are added to its build path"